### ISSUE TITLE
HTTP Listener rules fail to be created all of a sudden

### ISSUE THREAD

**Issue by str11ngfello** [aws/copilot-cli#3841]
Been using copilot version 1.18.1 for the last 4 months with no upgrades to the cli ever.

Yesterday I deployed this "dev" environment just fine (which has been in existence for months). Today, I'm deploying a one line code change and get this error (image below.) There haven't been any changes to infrastructure via copilot or in the aws console. I inspected the Load Balancer for the failing dev environment and the rules look identical to 2 other environments that are working. 

Trail is cold for me. Thoughts? I can't really decipher the error. S3 Bucket? I'm not using any S3 buckets perhaps it's one copilot uses. How can I track this down. I'm on the urgent side as I deploy to production in the morning :D.

Thank you!

**Comment by str11ngfello** [comment]
Here is the CloudFormation event log for a little more context. I'm afraid it doesn't give any additional clues to me, you?

![image](https://user-images.githubusercontent.com/877000/182437565-06b7c012-33a1-4f02-8d25-3d575e5a3301.png)

**Comment by str11ngfello** [comment]
Another service (api) in the same dev environment deploys just fine. In fact I was deploying it at the same time I was deploying the "app" service when it failed. I don't think theres an issue with simultaneous deployment. (?) The app and the api services  are both Load Balanced Web Service types. 

![image](https://user-images.githubusercontent.com/877000/182444685-4a2bd760-5bee-4fa1-b15a-71f248dd6fa2.png)

**Comment by Lou1415926** [comment]
Hello @str11ngfello ! I'm really sorry for the issue. Would you mind helping me with the following questions?

1. It seems like you are running copilot v1.20, is this correct?
2. Did you change your `http.path` and deploy?

**Comment by str11ngfello** [comment]
Hi @Lou1415926 actually I'm running 1.18.1 and have been for 3 months or so. I never upgraded from 1.18.1

I did not change my http.path, it's always been set the same for 3 months.

**Comment by iamhopaul123** [comment]
Hello @str11ngfello. Just wanted to double check: it seems like yesterday you successfully deployed the service to `dev` with no issue. And today you only made one line code changes and when deploying to the `dev` env again, you had the above issue. And you always use Copilot `v1.18.1`. Is that correct? If so would you mind sharing more CFN logs (would be nice if it can be before 08/02 11:23 update) for the `taunt-dev-app` stack?

**Comment by str11ngfello** [comment]
Hi @iamhopaul123 , yes those details are correct. I didn't know if there was a text export so I screen capped yesterday and today's run which resulted in 4 screenshots in chronological order. The 8/1 deploy you can see worked, then it heads into the 8/2 one today which ends in the error.

**Comment by str11ngfello** [comment]
Oh I feel silly, I used describe stacks command in the aws cli ... see attached log file

@iamhopaul123 

[stacklogs.txt](https://github.com/aws/copilot-cli/files/9245862/stacklogs.txt)
.

**Comment by iamhopaul123** [comment]
Hey @str11ngfello. It looks like from your yesterday's log, `RulePriorityFunctionRole` was created but we hadn't introduced that change until `v1.19`. Most likely yesterday you used a newer version than `v1.18.1` to deploy, which was successful because Copilot is backward compatible. However, today when you tried to use `v1.18.1` Copilot to deploy a service that was lastly deployed via a newer version, it's not gonna work since we are not forward compatible (aka older version doesn't know what newer version will change).

In short, could you double check if you have two version of Copilot installed locally or use multiple different machines to deploy? And in order to solve the problem, the simplest way might be just upgrade the Copilot CLI version to the latest and run `env deploy` to upgrade to the latest environment (because after v1.20 we won't upgrade the environment automatically) and then try `svc deploy` again.

**Comment by str11ngfello** [comment]
@iamhopaul123  Whoa that's wild. I have verified I have 1.18.1 and that's all I ever had. HOWEVER, there's one other developer and he has 1.20.0 BUT, he's working on a separate copilot application! So I work on the copilot app "taunt" and he works on the copilot app "tauntbattleworld".  I just confirmed with him he's only got access to tauntbattlworld.

See where I'm going with this? Hm, I wonder if we can see who deployed from the logs? Or, can 2 separate copilot applications affect each other in the same account? I know they are separate VPCs and load balancers and such, but the only explanation of a different version would be on his machine. So wild.

Let me keep thinking. Let me know if you have any thoughts on this front. Thank you for the quick replay... gonna be a late night for me!

We have two apps. He works on tauntbattleworld and I work on taunt.  We're in the same account. I wonder if there's a wire crossed between us.

![image](https://user-images.githubusercontent.com/877000/182485749-1394c56a-f8bc-45b8-8c21-bc55dc0761f5.png)

**Comment by iamhopaul123** [comment]
Hello @str11ngfello. I don't think two applications could affect each other, even if they are in the same account. And from the CFN console logs it seems like at least for yesterday's deployment you were using Copilot `v1.19^`. Maybe you have two different versions of Copilot installed?

**Comment by str11ngfello** [comment]
@Lou1415926  @iamhopaul123  Thank you for your help. Not only have you pointed us to our solution. You've saved our production deployment schedule. We were using two different copilot versions and he accidentally was on a branch in which the "taunt" app was specified in the workspace. So yes, he pushed yesterday, not me, and in the process upgraded us to 1.20.  We're an early project and CI/CD will be a great upgrade to our manual deployment process.

Because of your tremendous response time, I'll be able to sleep tonight. Thank you for the wonderful tool and support. Your team is amazing. Please don't stop copilot development. I come from a 4 year, heavy, Google K8s background and have loved the simplicity of copilot. :D

**Comment by huanjani** [comment]
We now have an enhancement that prevents this type of version downgrade situation in v1.29.0: https://github.com/aws/copilot-cli/releases/tag/v1.29.0!
